mutiny-core/src/node.rs

```diff
@@ -1148,6 +1034,139 @@ impl Node {
     }
 }
 
+async fn start_reconnection_handling(
+    multi_socket: &MultiWsSocketDescriptor,
+    websocket_proxy_addr: String,
+    self_connection: PubkeyConnectionInfo,
+    peer_man: Arc<dyn PeerManager>,
+    logger: &Arc<MutinyLogger>,
+    uuid: String,
+    lsp_client: &Option<LspClient>,
+) -> Result<(), MutinyError> {
+    let mut multi_socket_reconnect = multi_socket.clone();
+    let websocket_proxy_addr_copy = websocket_proxy_addr.clone();
+    let self_connection_copy = self_connection.clone();
+    spawn_local(async move {
+        loop {
+            if !multi_socket_reconnect.connected() {
+                debug!("got disconnected from multi socket proxy, going to reconnect");
+                match WsProxy::new(&websocket_proxy_addr_copy, self_connection_copy.clone()).await {
+                    Ok(main_proxy) => {
+                        multi_socket_reconnect.reconnect(Arc::new(main_proxy)).await;
+                    }
+                    Err(_) => {
+                        sleep(5 * 1000).await;
+                        continue;
+                    }
+                };
+            }
+            sleep(5 * 1000).await;
+        }
+    });
+
+    // Attempt connection to LSP first
```

Comment:
you have here we are connecting to the LSP first but we're actually connecting to the m2m proxy first. Would probably make sense to actually do the LSP first so that is ready as soon as possible.

Reply:
Moved to the top